groq.gemspec

```diff
@@ -33,7 +33,7 @@ Gem::Specification.new do |spec|
   spec.executables = spec.files.grep(%r{\Aexe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "faraday", "~> 2.0"
+  spec.add_dependency "faraday", "~> 1.0"
```

Comment:
Can we make this ">= 1.0" to allow 2.X as well?

Reply:
Any ideas how we might test both faraday 1.X and faraday 2.X in CI?

Reply:
Hi thank you! Yes I have updated this CR to allow for 2.x as well ✅ 

As for the latter I will push a new commit shortly editing main.yml workflow in a minute. The approach would be, IMO, to have the faraday version pulled from env var. In your workflow if the "FARADAY_VERSION" var is not set, it will default to the latest version as per Gemfile; if it is set it will pull the custom version.

```
    - name: Run the test suite with lower Faraday versions
      run: FARADAY_VERSION=1.0.0 bundle exec rake test
```

Reply:
Let me know if this works / makes sense - I cannot run CI without workflow approval so I hope this helps :)